src/ui/components/Pagination.tsx

```diff
@@ -5,14 +5,20 @@ export async function Pagination({
 	pageInfo,
 }: {
 	pageInfo: {
-		endCursor?: string | null;
+		baseUrl: string;
 		hasNextPage: boolean;
+		endCursor?: string | null;
+		urlSearchParams?: URLSearchParams;
 	};
 }) {
+	if (pageInfo.endCursor) {
+		pageInfo.urlSearchParams?.set("cursor", pageInfo.endCursor);
```

Comment:
This line mutates `pageInfo.urlSearchParams` – a parameter coming from the props. That can lead to unexpected results.